### PR TITLE
[RPD-233] Change the matcha-ml global variable config location from a users home folder

### PR DESCRIPTION
**Pull request by JonoCX**
This PR makes Matcha POSIX compliant. More specifically, the prior location of the global configuration file was in a `.matcha-ml` directory in the users home directory - this goes against POSIX principles. In this PR, that location is changed in the code to the `.config` directory in the home directory and `.matcha-ml` is no longer a hidden directory.

## Checklist

Please ensure you have done the following:

* [x] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [ ] I have updated the documentation if required.
* [x] I have added tests which cover my changes.

## Type of change

Tick all those that apply:

* [ ] Bug Fix (non-breaking change, fixing an issue)
* [ ] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [x] Other (add details above)

### CODE REVIEW

**Review by dudeperf3ct — APPROVED** [review]
Awesome work @JonoCX. LGTM

File: src/matcha_ml/services/global_parameters_service.py [hunk 1]
```diff
@@ -119,8 +119,9 @@ def default_config_file_path(self) -> str:
         Returns:
             The default global configuration directory.
         """
-        home_path = os.path.expanduser("~")
-        return os.path.join(home_path, ".matcha-ml", "config.yaml")
+        config_path = os.path.join(os.path.expanduser("~"), ".config")
+
+        return os.path.join(config_path, "matcha-ml", "config.yaml")
```
Comment (dudeperf3ct) [inline src/matcha_ml/services/global_parameters_service.py:124]: Does `config_file` function create a config folder if not present?

File: tests/test_services/test_global_parameters_service.py [hunk 2]
```diff
@@ -29,7 +42,9 @@ def test_class_is_singleton(matcha_testing_directory):
         new_callable=mock.PropertyMock,
     ) as file_path:
         file_path.return_value = str(
-            os.path.join(str(matcha_testing_directory), ".matcha-ml", "config.yaml")
+            os.path.join(
```
Comment (swells2020) [inline tests/test_services/test_global_parameters_service.py:45]: Can we use the fixture defined above here instead?

**Review by swells2020 — APPROVED** [review]
💪 it's great. One small comment.

File: src/matcha_ml/services/global_parameters_service.py (hunk 1 above)
Comment (JonoCX) [reply src/matcha_ml/services/global_parameters_service.py:124]: Yep, in this line: https://github.com/fuzzylabs/matcha/blob/72003c2dda40b663c8ea64202a6f820e9bbdcc4f/src/matcha_ml/services/global_parameters_service.py#L65

**Review by swells2020 — APPROVED** [review]
LGTM 🚀

**Pull request merged**